README.md

```diff
@@ -90,116 +66,4 @@ As of version 0.11.0, all interfaces will be required to be  `Send+Sync`, and th
 `[Threadsafe]` attribute will be deprecated and ignored.
 
 See also [adr-0004](https://github.com/mozilla/uniffi-rs/blob/main/docs/adr/0004-only-threadsafe-interfaces.md)
-which outlines the reasoning behind this decision.
-
-### Component Interface
```

Comment:
These details on "how does UniFFI work" are mostly duplicated in the user guide, so I deleted them from here.